### ISSUE TITLE
adapt date format with defined project's locale

### ISSUE THREAD

**Issue by didntchooseaname** [retypeapp/retype#573]
Good morning, and thank you for retype.

1. Is it possible to adapt the date format with locale please ?
Like for the "fr" one:
instead of 2022-03-04 -> 04-03-2022.

2. there is a warning when i build my project (locale defined in the retype.yml file), it basically say that "fr" locale isn't recognized (en is).
![warning](https://github.com/retypeapp/retype/assets/126179594/aca1043c-357f-4009-9c2b-69476ca88f8c)
But in facts the setting works for the whole project.
(project: https://docs.contactit.fr)

Thank you,

**Comment by didntchooseaname** [comment]
![date](https://github.com/retypeapp/retype/assets/126179594/65b9f9b7-f1c9-4f18-9628-29ebe3ca48f5)
Like that with "fr" locale

**Comment by geoffreymcgill** [comment]
> 2. there is a warning when i build my project (locale defined in the retype.yml file), it basically say that "fr" locale isn't recognized (en is).

I was able to reproduce the warning message. We will investigate and figure out why this is warning is happening. Best I can tell, the warning should not be happening. 

```
WARNING: [retype.yml] Not supported value: "fr" in "locale".
```

**Comment by geoffreymcgill** [comment]
> ![date](https://user-images.githubusercontent.com/126179594/247855977-65b9f9b7-f1c9-4f18-9628-29ebe3ca48f5.png)
> Like that with "fr" locale

I'm assuming that screen capture is created using the `date` page config?

```yml
date: 2022-09-09
```

**Comment by didntchooseaname** [comment]
> > ![date](https://user-images.githubusercontent.com/126179594/247855977-65b9f9b7-f1c9-4f18-9628-29ebe3ca48f5.png)
> > Like that with "fr" locale
> 
> I'm assuming that screen capture is created using the `date` page config?
> 
> ```yaml
> date: 2022-09-09
> ```

Yes in the md file of the page. When i change it like: 2022-03-04 -> 04-03-2022
There a warning as well:
![date_warning](https://github.com/retypeapp/retype/assets/126179594/8050cae4-f772-476d-a470-e2a8a7212f8e)

**Comment by geoffreymcgill** [comment]
> Yes in the md file of the page. When i change it like: 2022-11-08 -> 08-11-2022

I assume you would like the date format to be `08-11-2022` in both the Page `date` config and what is printed to the page, such as `Publié 08-11-2022`?

**Comment by didntchooseaname** [comment]
> > Yes in the md file of the page. When i change it like: 2022-11-08 -> 08-11-2022
> 
> I assume you would like the date format to be `08-11-2022` in both the Page `date` config and what is printed to the page, such as `Publié 08-11-2022`?

yes exactly.

**Comment by geoffreymcgill** [comment]
> yes exactly.

The `date` config requires a date value in an [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) international standard using the format of `YYYY-MM-DD`. Personally, I feel using a customizable or locale specific format for the `date` config is not a good idea. The `date` value is data and should be configured in an international standard.

There is an argument to be made that the output should be locale specific, so for `locale: fr`, the format that is printed to the page could be automatically converted to the `DD-MM-YYYY` format. Currently, there are no configs or values within the [localization files](https://github.com/retypeapp/retype-translations/blob/main/src/UserInterface.fr.resx) that allow for this locale specific date format. Maybe even a project level config for `locale`, such as:

```yml
locale:
  - lang: fr
  - dateFormat: DD-MM-YYYY
```

We will look into what it will take to add a `dateFormat` to the locale files and/or as a project level config.

**Comment by geoffreymcgill** [comment]
I have a question for you...

In the page https://docs.contactit.fr/reseaux/alcatel/, you have the following configuration:

https://github.com/didntchooseaname/docs.contactit.fr/blob/master/Reseaux/Alcatel/index.md?plain=1#L2-L13C45

```yml
---
title: Configuration d'un switch Alcatel
icon: three-bars
description: Commandes utiles pour switch Alcatel.
categories: [Reseaux]
date: 2021-11-12
tags: [alcatel, networks]
visibility: public
order: 99
locale: fr
---

# :gloves: Configuration d'un switch Alcatel
```


But, I see you have both the `title` and a `# :gloves: Configuration d'un switch Alcatel` heading configured. 

How are you getting only one of those to be rendered on the page?

When I run that page, I see both, as demonstrated in the following screen capture:

<img width="837" alt="Screenshot 2023-06-22 at 2 19 29 AM" src="https://github.com/retypeapp/retype/assets/62210/658571d0-8972-47c8-af4f-c405332980bc">

**Comment by geoffreymcgill** [comment]
Setting the `title`, `visibility: public`, and `locale: fr` on each of the pages is not required. 

The `title` is not required if you add a page `#` title heading, such as `# :gloves: Configuration d'un switch Alcatel`. Use one or the other, but not both, and ideally only use the page title heading (`# Page TItle Here`) instead of the `title` config. 

The config `visibility: public` is the default value, so you can completely remove that line from all pages. 

The config `locale: fr` is project level config, so it will only work in your project `retype.yml` file, and that line can be removed from all pages.

**Comment by geoffreymcgill** [comment]
> How are you getting only one of those to be rendered on the page?

I think I might know why this is happening on my machine. I will have to run some more tests to try and isolate the exact reason why.

**Comment by didntchooseaname** [comment]
> Setting the `title`, `visibility: public`, and `locale: fr` on each of the pages is not required.
> 
> The `title` is not required if you add a page `#` title heading, such as `# :gloves: Configuration d'un switch Alcatel`. Use one or the other, but not both, and ideally only use the page title heading (`# Page TItle Here`) instead of the `title` config.
> 
> The config `visibility: public` is the default value, so you can completely remove that line from all pages.
> 
> The config `locale: fr` is project level config, so it will only work in your project `retype.yml` file, and that line can be removed from all pages.

Thank you i deleted these points. i tried with  locale fr on pages and after i understood that it was at the project's level when i read the doc.

**Comment by geoffreymcgill** [comment]
The issue that triggered the `Not supported value: "fr" in "locale"` warning has been fixed. The original logic did not appear to affect the building or functionality of the website, so for now, you can just ignore that message. 

The fix will be included in the upcoming v3.1 release.

**Comment by didntchooseaname** [comment]
> I'm going to reopen this issue as we do intend to implement a customizable `dateFormat` project config. The new feature will be included as part of a series of locale related features.
> 
> A timeline for release is not available, but I will keep this thread updated with our progress.

Okay thanks you, glad to see the changes in the releases :)
